cameraServer.py

```diff
@@ -115,31 +168,46 @@ def getRetroPos(img, display=False, distance_away=distance_away):
     cameraConfigs = readConfig()
 
     # start NetworkTables
-    NetworkTables.initialize(server='10.47.74.2')
+    NetworkTables.initialize(server="10.47.74.2")
 
     NetworkTables.setUpdateRate(1)
-    nt = NetworkTables.getTable('/vision')
-    entry_tape_angle = nt.getEntry('target_tape_error')
-    entry_game_piece = nt.getEntry('game_piece')
-    entry_outake = nt.getEntry('within_deposit_range')
+    nt = NetworkTables.getTable("/vision")
+    ping = nt.getEntry("ping")
+    raspi_pong = nt.getEntry("raspi_pong")
+    rio_pong = nt.getEntry("rio_pong")
+
+    entry_game_piece = nt.getEntry("game_piece")
+    entry_dist = nt.getEntry("fiducial_x")
+    entry_offset = nt.getEntry("fiducial_y")
+    entry_fiducial_time = nt.getEntry("fiducial_time")
 
     # start cameras
     cameras = []
     for cameraConfig in cameraConfigs:
         cameras.append(startCamera(cameraConfig))
 
-    
     cargo_sink = cameras[0][0].getVideo(camera=cameras[0][1])
     hatch_sink = cameras[1][0].getVideo(camera=cameras[1][1])
 
-    source = cameras[0][0].putVideo('Driver_Stream', 320, 240)
-    source2 = cameras[1][0].putVideo('mask', 320, 240)
+    source = cameras[0][0].putVideo("Driver_Stream", screenX, screenY)
+    # source2    = cameras[1][0].putVideo('mask', 320, 240)
 
-    frame = np.zeros(shape=(screenSize[1], screenSize[0], 3))
-    game_piece = 0 #0 = hatch, 1 = cargo
+    frame = np.zeros(shape=(screenSize[1], screenSize[0], 3), dtype=np.uint8)
+    image = np.zeros(shape=(screenSize[1], screenSize[0], 3), dtype=np.uint8)
+    hsv = np.zeros(shape=(screenSize[1], screenSize[0], 3), dtype=np.uint8)
+    mask = np.zeros(shape=(screenSize[1], screenSize[0], 3), dtype=np.uint8)
+    img = np.zeros(shape=(screenSize[1], screenSize[0], 3), dtype=np.uint8)
 
+    game_piece = 0  # 0 = hatch, 1 = cargo
+    old_ping_time = 0
     while True:
+        ping_time = ping.getNumber(0)
+        if abs(ping_time - old_ping_time) > 0.00000001:
+            raspi_pong.setNumber(time.monotonic())
+            rio_pong.setNumber(ping_time)
+            old_ping_time = ping_time
         game_piece = entry_game_piece.getBoolean(0)
+        fiducial_time = time.monotonic()
         if not game_piece:
             frame_time, frame = hatch_sink.grabFrameNoTimeout(image=frame)
```

Comment:
In your handling of the two gamepieces you are writing the exact same code, a more succinct way would be to use a turnery expression to determine which camera you are using then get the frame and frame time.  Something along the lines of `camera = cargo_sink if game_piece else hatch_sink`.

Reply:
> https://www.thefreedictionary.com/turnery

TIL a new word. Somehow I don't think it's the word you're looking for.

Reply:
Sorry meant ternary, thanks Davo